solr/core/src/java/org/apache/solr/request/SolrRequestInfo.java

```diff
@@ -38,7 +40,13 @@
 
 
 public class SolrRequestInfo {
-  protected final static ThreadLocal<SolrRequestInfo> threadLocal = new ThreadLocal<>();
+
+  protected final static int capacity = 150;
+
+  protected final static ThreadLocal<Deque<SolrRequestInfo>> threadLocal = ThreadLocal.withInitial(() -> {
+    Deque<SolrRequestInfo> stack = new ArrayDeque<>(capacity);
```

Comment:
I think you can simply do a lambda reference to ArrayDequeue::new.  No need initialize with an internal capacity.  Honestly I'd prefer a LinkedList because in practice this stack will be extremely small (zero, one, sometimes two, very unlikely more).  But I leave that impl choice to you.